src/components/FilmCardComponent/FilmCard.jsx

```diff
@@ -1,35 +1,73 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './FilmCard.module.css';
+import ApiClient from '../../api/apiClient';
 
-const FilmCard = (props) => {
-    return (
-        <div
-            className={`card ${styles.filmCard}`}
-            style={{ width: 18 + 'rem' }}
-        >
-            <img
-                src={'poster_path'}
-                className={`card-img-top ${styles.filmCardImg}`}
-                alt="poster"
-            />
-            <div className={`card-body ${styles.filmCardBody}`}>
-                <h5 className={`card-title ${styles.filmCardTitle}`}>
-                    {props.title}
-                </h5>
-                <p className={`card-text ${styles.filmCardYear}`}>
-                    {props.release_date}
-                </p>
-                <p className={`card-text ${styles.filmCardGenresList}`}></p>
+const api = new ApiClient();
+
+class FilmCard extends React.Component {
+    state = {
+        genres: null,
+    };
+
+    componentDidMount() {
+        this.getGenres(this.props.genres);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.genres !== this.props.genres) {
+            this.getGenres(this.props.genres);
+        }
+    }
+
+    getGenres = (ids) => {
+        api.getAllGenres().then((res) => {
+            let genres = '';
+            res.genres.forEach((genre) => {
```

Comment:
in this case, you can use res.genres.filter.

Reply:
Fixed